tests/tcp.rs

```diff
@@ -26,6 +26,68 @@ async fn bind() -> std::result::Result<TcpListener, std::io::Error> {
     TcpListener::bind((IpAddr::from(Ipv4Addr::UNSPECIFIED), PORT)).await
 }
 
+#[test]
+fn connects_within_a_localhost() -> Result {
+    let mut sim = Builder::new().build();
+
+    sim.host("server", || async {
```

Comment:
You could simplify this by using `sim.client` and that is guaranteed to run to completion on `sim.run()`.